src/components/Form/Textarea.tsx

```diff
@@ -0,0 +1,139 @@
+import { Paragraph } from '../Paragraph/Paragraph';
+import classnames from 'classnames';
+import styled from 'styled-components';
+import { idGenerator } from '../../utils/uuid';
+
+const baseClassName = 'fi-textarea';
+const textareaClassNames = {
+  label: `${baseClassName}_label`,
+  optionalText: `${baseClassName}_optionalText`,
+  textarea: `${baseClassName}_textarea`,
+  hintText: `${baseClassName}_hintText`,
+  statusText: `${baseClassName}_statusText`,
+};
+
+export interface TextareaProps extends HtmlTextareaProps {
+  /** Custom classname to extend or customize */
+  className?: string;
+  /** Disable usage */
+  disabled?: boolean;
+  /** Event handler to execute when clicked */
+  onClick?: () => void;
+  /** To execute on textarea text change */
+  onChange?: (event: ChangeEvent<HTMLTextAreaElement>) => void;
+  /** To execute on textarea text onBlur */
+  onBlur?: (event: FocusEvent<HTMLTextAreaElement>) => void;
+  /** Label */
+  labelText: string;
+  /** Hide or show label. Label element is always present, but can be visually hidden.
+   * @default visible
+   */
+  labelMode?: 'hidden' | 'visible';
+  /** Placeholder text for input. Use only as visual aid, not for instructions. */
+  visualPlaceholder?: string;
+  /** Text content for textarea */
+  children?: string;
+  /** Hint text to be shown below the component */
+  hintText?: string;
+  /**
+   * 'default' | 'error'
+   * @default default
+   */
+  status?: 'default' | 'error';
+  /** Status text to be shown below the component and hint text. Use e.g. for validation error */
+  statusText?: string;
+  /** Resize mode of the textarea
+      'both' | 'vertical' | 'horizontal' | 'none'
+      @default 'vertical' 
+   */
+  resize?: 'both' | 'vertical' | 'horizontal' | 'none';
+  optionalText?: string;
+}
+
+class BaseTextarea extends Component<TextareaProps> {
+  render() {
+    const {
+      id: propId,
+      className,
+      disabled = false,
+      children,
+      onClick,
+      labelMode,
+      labelText,
+      hintText,
+      status,
+      statusText,
+      visualPlaceholder,
+      resize: dismissResize,
+      optionalText,
+      ...passProps
+    } = this.props;
+
+    const onClickProps = !!disabled ? {} : { onMouseDown: onClick };
+
+    const hideLabel = labelMode === 'hidden';
+    const id = idGenerator(propId);
+    const statusTextId = `${idGenerator(propId)}-statusText`;
+    const hintTextId = `${idGenerator(propId)}-hintText`;
+
+    return (
+      <HtmlLabel className={classnames(baseClassName, className, {})}>
+        {hideLabel ? (
+          <VisuallyHidden>{labelText}</VisuallyHidden>
+        ) : (
+          <Paragraph>
+            <HtmlSpan className={textareaClassNames.label}>
+              {labelText}
+            </HtmlSpan>
+            {optionalText && (
+              <HtmlSpan className={textareaClassNames.optionalText}>
+                {`(${optionalText})`}
```

Comment:
Is string interpolation necessary here?

Reply:
Not necessarily, but when changing it to: `({optionalText})`. Then our eslint rules will complain: `{optionalText} must be placed on a new line eslintreact/jsx-one-expression-per-line`
And if trying to put them on new line, autoformat will change it back to the starting state `({optionalText})`

And in the DOM it will show like this:
![image](https://user-images.githubusercontent.com/53757053/89185201-84580280-d5a2-11ea-96a2-65d36ba1e52d.png)

But it does look same visually for the user though.

Reply:
Ok, we have bad linter rules 😓  In this case template literal solution is better in my opinion too.